src/services/db/GitFileSystemService.ts

```diff
@@ -737,6 +792,49 @@ export default class GitFileSystemService {
     })
   }
 
+  mergeStagingToMaster(
+    repoName: string
+  ): ResultAsync<boolean, GitFileSystemError> {
+    const efsVolPath = this.getEfsVolPathFromBranch("staging")
+    return this.isValidGitRepo(repoName, "staging").andThen((isValid) => {
+      if (!isValid) {
+        return errAsync(
+          new GitFileSystemError(
+            `Folder "${repoName}" for EFS vol path: "${efsVolPath}" is not a valid Git repo`
+          )
+        )
+      }
+
+      return (
+        this.ensureCorrectBranch(repoName, "master")
+          .andThen(() =>
+            ResultAsync.fromPromise(
+              this.git
+                .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+                .merge(["staging", "--no-ff"]), // ensure there is a merge commit
```

Comment:
could you add comments on what -no-ff is doing?

Reply:
added the comment, but removing this function from this PR for now as its unused and was confusing to others
[27d6435](https://github.com/isomerpages/isomercms-backend/pull/1172/commits/27d643548cec3739e018faa88a04e97a76e6e23a)